RecoMuon/L2MuonSeedGenerator/src/L2MuonSeedGeneratorFromL1T.cc

```diff
@@ -124,246 +157,804 @@ void L2MuonSeedGeneratorFromL1T::produce(edm::Event& iEvent, const edm::EventSet
 
-          if (!detsWithStates.empty()){
-  
-            TrajectoryStateOnSurface newTSOS = detsWithStates.front().second;
-            const GeomDet *newTSOSDet = detsWithStates.front().first;
-    
-            LogTrace(metname) << "Most compatible det";
-            LogTrace(metname) << debug.dumpMuonId(newTSOSDet->geographicalId());
-  
-            LogDebug(metname) << "L1 info: Det and State:";
-            LogDebug(metname) << debug.dumpMuonId(newTSOSDet->geographicalId());
-  
-            if (newTSOS.isValid()){
-  
-              //LogDebug(metname) << "(x, y, z) = (" << newTSOS.globalPosition().x() << ", " 
-              //                  << newTSOS.globalPosition().y() << ", " << newTSOS.globalPosition().z() << ")";
-              LogDebug(metname) << "pos: (r=" << newTSOS.globalPosition().mag() << ", phi=" 
-                                << newTSOS.globalPosition().phi() << ", eta=" << newTSOS.globalPosition().eta() << ")";
-              LogDebug(metname) << "mom: (q*pt=" << newTSOS.charge()*newTSOS.globalMomentum().perp() << ", phi=" 
-                                << newTSOS.globalMomentum().phi() << ", eta=" << newTSOS.globalMomentum().eta() << ")";
-  
-              //LogDebug(metname) << "State on it";
-              //LogDebug(metname) << debug.dumpTSOS(newTSOS);
-  
-              const TrajectorySeed *assoOffseed = 
-                associateOfflineSeedToL1(offlineSeedHandle, offlineSeedMap, newTSOS, dRcone);
-  
-              if(assoOffseed!=nullptr) {
-                PTrajectoryStateOnDet const & seedTSOS = assoOffseed->startingState();
-                TrajectorySeed::const_iterator 
-                  tsci  = assoOffseed->recHits().first,
-                  tscie = assoOffseed->recHits().second;
-                for(; tsci!=tscie; ++tsci) {
-                  container.push_back(*tsci);
+            if (detsWithStates.empty() && barrel ) {
+              // try again to propagate but using ME2 as reference
+              tsos = theService->propagator(thePropagatorName)->propagate(state, ME2DetLayer->surface());
+              detsWithStates = ME2DetLayer->compatibleDets(tsos, 
+                                             *theService->propagator(thePropagatorName), 
+                                             *theEstimator);   
+            }
+
+            if (!detsWithStates.empty()){
+
+              TrajectoryStateOnSurface newTSOS = detsWithStates.front().second;
+              const GeomDet *newTSOSDet = detsWithStates.front().first;
+
+              LogTrace(metname) << "Most compatible det";
+              LogTrace(metname) << debug.dumpMuonId(newTSOSDet->geographicalId());
+
+              LogDebug(metname) << "L1 info: Det and State:";
+              LogDebug(metname) << debug.dumpMuonId(newTSOSDet->geographicalId());
+
+              if (newTSOS.isValid()){
+
+                //LogDebug(metname) << "(x, y, z) = (" << newTSOS.globalPosition().x() << ", "
```

Comment:
What the reason for letting the LogDebugs commented out (here and below)?
Please either remove (better) or uncomment